numerical/interpolation.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+from numerical import splines
+from numerical.utils.interpolation import repeat_args
+
+
+def interpolate(values, grid, batch_size=16):
+    """ Builds function which is an interpolation function on nodes with computer values in these nodes.
+
+    Args:
+        values: list of function values in grid nodes.
+        grid: points where function was calculated used np.meshgrid function with parameter 'indexing='ij''.
+        batch_size: int, batch size for interpolation process.
+
+    Returns:
+        interpolated function.
+    """
+    # linear interpolation will be used as a basis function
+    bfunc = splines.linear
+
+    nodes_range = [(g.min(), g.max()) for g in grid]
+    nodes_count = grid[0].shape
+    nodes_dim = len(grid)
+    shift_indexes = [np.arange(0, dim, dtype=np.float64) for dim in nodes_count]
+    values = values.ravel()
+
+    def _interpolated(x):
+        """ Interpolated function.
+
+        Args:
+            x: numpy.ndarray
+
+        Returns:
+            numpy.ndarray
+        """
+
+        if x.shape[-1] != 1:
+            x = np.expand_dims(x, axis=-1)
+
+        result = []
+        batch_position = 0
+        _interpolation_loop(result, x, values, bfunc, nodes_range, nodes_count,
```

Comment:
Return `result` from the function.
Too many arguments.

Reply:
this function has the same structure as `integrate`. Since `integrate` uses recursion, it is convenient to modify argument `result` instead of returning it.

Reply:
Okay